### ISSUE TITLE
No detections reported in data after training

### ISSUE THREAD

**Issue by PranjaLBiswas27** [AlexeyAB/darknet#1571]
Hi,

I tried to train YOLOv3 on my custom data which I labelled using yolo_mark. I trained the network for 2000 iterations only, cause somehow avg loss became 0, but all other field were nan.  So I dont really know if the network got trained or not.  But still I tried to use the weights to detect my objects, but i ended with the program just showing me the image itself without any detections. What exactly went wrong? Help will be greatly appreciated. Thanks a lot in advance.

Kind Regards
Pranjal Biswas

**Comment by derekwong66** [comment]
decrease learning rate, and increase iterations, you should set step as the number of  classes * 2000

**Comment by AlexeyAB** [comment]
@PranjaLBiswas27 

* What mAP can you get?
* What avg loss did you get at the end of training?
* How many classes and images do you use?

**Comment by PranjaLBiswas27** [comment]
@AlexeyAB 
1. I get a map of 0, on running the following command
./darknet detector map data/obj.data yolo-obj.cfg backup/yolo-obj_2000.weights

2. I force stopped traing after about 2070 iterations, and that time avg loss was 0 in the graph.

3. I have 3 classes in total, with around 600 images containing all 3 classes, and around 100 images with 2 classes only. So in total around 700 images in the training set. Another 150 images in the test set.

**Comment by AlexeyAB** [comment]
So look at the output log during training, are there some error messages.
Also look at the file `bad.list` and `bad_label.list` if they exist.

It looks like that your cfg-file or your dataset is wrong.

Can you attach your cfg-file to your message?

**Comment by PranjaLBiswas27** [comment]
Hi @AlexeyAB ,

I deleted everything and tried following your steps again. I think earlier I made a mistake in the location of the files that should have been in the "build/Darknet/x64" directory of darknet-master, instead i put them in "darknet-master" itself. Now, YOLOv3 works fine and gives me 90% map with 0.403 avg loss.

Thanks a lot for this awesome work and maintenance of this repository, also for helping with all my queries all throughout the process. Greatly appreciated.

Kind Regards
Pranjal Biswas
